resources-domain/resources-import-producer/transformExternalResourceToApiResource.ts

```diff
@@ -0,0 +1,212 @@
+    }
+
+    aseloResource.attributes.ResourceNumberAttributes.push({
+      key: mapping.keyGenerator(context),
+      value,
+      info: mapping.infoGenerator(context),
+    });
+  } else if (mapping.table === 'ResourceDateTimeAttributes') {
+    const value = mapping.valueGenerator(context);
+    if (typeof value !== 'string') {
+      throw new Error(`Wrong value provided to ResourceDateTimeAttributes: mapping ${JSON.stringify(mapping)} and value ${value}`);
+    }
+
+    aseloResource.attributes.ResourceDateTimeAttributes.push({
+      key: mapping.keyGenerator(context),
+      value,
+      info: mapping.infoGenerator(context),
+    });
+  } else {
+    throw new Error(`Unhandled case for provided mapping: mapping ${JSON.stringify(mapping)}`);
+  }
+};
+
+/**
+ * Function to map a KHP node into an Aselo resource.
+ * This function mutates the given resource.
+ * Recusively transform the children (if any).
+ * @param mappingNode
+ * @param dataNode
+ * @param parentContext
+ * @param aseloResource
+ * @returns
+ */
+const mapNode = (
+  mappingNode: MappingNode,
+  dataNode: any,
+  parentContext: FieldMappingContext,
+  aseloResource: ImportApiResource,
+): ImportApiResource => {
+  Object.entries(mappingNode).forEach(([property, { children, ...mapping }]) => {
+    const captureProperty = property.match(/{(?<captureProperty>.*)}/)?.groups?.captureProperty;
+    const dataProperties: string[] = (captureProperty ? Object.keys(dataNode) : [property])
+      // Escape forward slashes in path segments
+      .map(p => p.replace('/', '\\/'));
+    dataProperties.forEach(dataProperty => {
+      const dataPropertyValue = dataNode[dataProperty];
+
+      // Node contains an empty value when we expected something to map, stop recursing
+      if (dataPropertyValue === null || dataPropertyValue === undefined) {
+        return;
+      }
+
+      const context: FieldMappingContext = {
+        ...parentContext,
+        currentValue: dataPropertyValue,
+        path: [...parentContext.path, dataProperty],
+      };
+      if (captureProperty) {
+        parentContext.captures[captureProperty] = dataProperty;
+      }
+
+      // Add the node to the corresponding resource table based on the specified mapping
+      if (isResourceFieldMapping(mapping)) {
+        pushResourceFieldMapping({ aseloResource, mapping, context });
+      } else if (isReferenceAttributeMapping(mapping)) {
+        pushReferenceAttributeMapping({ aseloResource, mapping, context });
+      } else if (isTranslatableAttributeMapping(mapping)) {
+        pushTranslatableAttributeMapping({ aseloResource, mapping, context });
+      } else if (isInlineAttributeMapping(mapping)) {
+        pushInlineAttributeMapping({ aseloResource, mapping, context });
+      }
+
+      // Recurse on the children node(s) if any
+      if (children) {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        const { _id, objectId, ...rest } = children;
```

Comment:
Do we want to use `objectId` annd/or `_id`? I was filtering them out here, as they are sibling of languages sometimes (objects like `{ en: ..., fr: ..., _id: 123 }`), but this needs to be removed if we want to make use of them as identifiers.